projects/igniteui-angular/src/lib/date-picker/date-picker.utils.ts

```diff
@@ -0,0 +1,730 @@
+                case FORMAT_DESC.NUMERIC: {
+                    inputMask.push('0000');
+                    break;
+                }
+                case FORMAT_DESC.TWO_DIGITS: {
+                    inputMask.push('00');
+                    break;
+                }
+            }
+        }
+
+        if (dateStruct[i].type === SEPARATOR) {
+            inputMask.push(dateStruct[i].value);
+        }
+    }
+
+    return inputMask.join('');
+}
+
+/**
+ *@hidden
+ */
+export function getMask(dateStruct: any[]): string {
+    const mask = [];
+    for (let i = 0; i < dateStruct.length; i++) {
+        switch (dateStruct[i].formatType) {
+            case FORMAT_DESC.NUMERIC: {
+                if (dateStruct[i].type === DATE_PARTS.DAY) {
+                    mask.push('d');
+                }
+                if (dateStruct[i].type === DATE_PARTS.MONTH) {
+                    mask.push('M');
+                }
+                if (dateStruct[i].type === DATE_PARTS.YEAR) {
+                    mask.push('yyyy');
+                }
+                break;
+            }
+            case FORMAT_DESC.TWO_DIGITS: {
+                if (dateStruct[i].type === DATE_PARTS.DAY) {
+                    mask.push('dd');
+                }
+                if (dateStruct[i].type === DATE_PARTS.MONTH) {
+                    mask.push('MM');
+                }
+                if (dateStruct[i].type === DATE_PARTS.YEAR) {
+                    mask.push('yy');
+                }
+            }
+        }
+
+        if (dateStruct[i].type === SEPARATOR) {
+            mask.push(dateStruct[i].value);
+        }
+    }
+
+    return mask.join('');
+}
+
+/**
+ *@hidden
+ */
+export function parseDateArray(dateFormatParts: any[], prevDateValue: Date, inputValue: string): any {
+    const dayStr = getDayValueFromInput(dateFormatParts, inputValue);
+    const monthStr = getMonthValueFromInput(dateFormatParts, inputValue);
+    const yearStr = getYearValueFromInput(dateFormatParts, inputValue);
+    const yearFormat = getDateFormatPart(dateFormatParts, DATE_PARTS.YEAR).formatType;
+    const day = (dayStr !== '') ? Number(dayStr) : 1;
+    const month = (monthStr !== '') ? Number(monthStr) - 1 : 0;
+
+    let year;
+    if (yearStr === '') {
+        year = (yearFormat === FORMAT_DESC.TWO_DIGITS) ? '00' : '2000';
+    } else {
+        year = yearStr;
+    }
+    let yearPrefix;
+    if (prevDateValue !== null && prevDateValue !== undefined) {
+        const originalYear = prevDateValue.getFullYear().toString();
+        if (originalYear.length === 4) {
+            yearPrefix = originalYear.substring(0, 2);
+        }
+    } else {
+        yearPrefix = '20';
+    }
+    const fullYear = (yearFormat === FORMAT_DESC.TWO_DIGITS) ? yearPrefix.concat(year) : year;
+
+    if ((month < 0) || (month > 11)) {
+        return { state: DATE_STATE.INVALID, value: inputValue };
+    }
+
+    if ((day < 1) || (day > daysInMonth(fullYear, month + 1))) {
```

Comment:
`day` could be NaN because of the conversion you're doing above. Add check for NaN.